charms/tensorboards-web-app/tests/unit/test_operator.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+from unittest.mock import MagicMock, patch
+
+import ops.testing
+import pytest
+import yaml
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus
+from ops.testing import Harness
+
+from charm import TensorboardsWebApp
+
+# Enable simulation of container networking
+ops.testing.SIMULATE_CAN_CONNECT = True
+APP_NAME = "tensorboards-web-app"
+
+
+@pytest.fixture(scope="function")
+def harness() -> Harness:
+    """Create and return Harness for testing."""
+    harness = Harness(TensorboardsWebApp)
+
+    # Set up container networking simulation
+    harness.set_can_connect(APP_NAME, True)
```

Comment:
Yep this holds, you can drop it: https://juju.is/docs/sdk/write-a-unit-test-for-a-charm#heading--simulate-can-connect

Reply:
Removing 
```
    # Set up container networking simulation
    harness.set_can_connect(APP_NAME, True)
 ```
actually results in all tests breaking with 
```
Traceback (most recent call last):
  File "/home/ubuntu/github/kubeflow-tensorboards-operator/charms/tensorboards-web-app/tests/unit/test_operator.py", line 38, in test_not_leader
    harness.begin_with_initial_hooks()
  File "/home/ubuntu/github/kubeflow-tensorboards-operator/charms/tensorboards-web-app/.tox/unit/lib/python3.8/site-packages/ops/testing.py", line 349, in begin_with_initial_hooks
    charm.on.config_changed.emit()
  File "/home/ubuntu/github/kubeflow-tensorboards-operator/charms/tensorboards-web-app/.tox/unit/lib/python3.8/site-packages/ops/framework.py", line 354, in emit
    framework._emit(event)
  File "/home/ubuntu/github/kubeflow-tensorboards-operator/charms/tensorboards-web-app/.tox/unit/lib/python3.8/site-packages/ops/framework.py", line 830, in _emit
    self._reemit(event_path)
  File "/home/ubuntu/github/kubeflow-tensorboards-operator/charms/tensorboards-web-app/.tox/unit/lib/python3.8/site-packages/ops/framework.py", line 919, in _reemit
    custom_handler(event)
  File "/home/ubuntu/github/kubeflow-tensorboards-operator/charms/tensorboards-web-app/src/charm.py", line 172, in _on_event
    self._check_container_connection()
  File "/home/ubuntu/github/kubeflow-tensorboards-operator/charms/tensorboards-web-app/src/charm.py", line 203, in _check_container_connection
    raise ErrorWithStatus(MaintenanceStatus("Waiting for container startup to complete"))
TypeError: __init__() missing 1 required positional argument: 'status_type'
```

Reply:
I believe the above comment is outdated, since you no longer use `_check_container_connection`. But in any case, the `TypeError` doesn't look directly related to you removing `set_can_connect`, if anything it was just now exposed.

The error should be raised like this (same way `update_layer` does it):
```
raise ErrorWithStatus("Waiting for container startup to complete", MaintenanceStatus)
```
Can you please rerun the tests using the current iteration on the charm?

Reply:
You 're right, removed it and now tests ran successfully so I pushed a commit with it!